source/common/http/conn_manager_impl.cc

```diff
@@ -1467,6 +1468,28 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ResponseHeaderMap& heade
 
   chargeStats(headers);
 
+  if (auto& proxy_status_config = connection_manager_.config_.proxyStatusConfig();
+      proxy_status_config.attach_proxy_status()) {
+    // Writing the Proxy-Status header is gated on |attach_proxy_status|.
+    // The |details| field and other internals are generated in
+    // fromStreamInfo().
+    if (absl::optional<StreamInfo::ProxyStatusError> proxy_status =
+            StreamInfo::ProxyStatusUtils::fromStreamInfo(filter_manager_.streamInfo());
+        proxy_status.has_value()) {
+      headers.setProxyStatus(StreamInfo::ProxyStatusUtils::toString(
```

Comment:
lazy question: for multi-envoy deployments should we overwrite or append?

Reply:
According to spec we should append.

Reply:
Great catch, done.